internal/service/service.go

```diff
@@ -0,0 +1,180 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/VTerenya/employees/internal"
+	"github.com/VTerenya/employees/internal/repository"
+)
+
+type Service interface {
+	CreatePosition(p *internal.Position) error
+	CreateEmployee(e *internal.Employee) error
+	GetPositions(limit, offset int) ([]internal.Position, error)
+	GetEmployees(limit, offset int) ([]internal.Employee, error)
+	GetPosition(id string) (internal.Position, error)
+	GetEmployee(id string) (internal.Employee, error)
+	DeletePosition(id string) error
+	DeleteEmployee(id string) error
+	UpdatePosition(p *internal.Position) error
+	UpdateEmployee(e *internal.Employee) error
+}
+
+type Serv struct {
+	repo *repository.Repo
+}
+
+func NewServ(repository *repository.Repo) *Serv {
+	return &Serv{
+		repo: repository,
+	}
+}
+
+func (t Serv) CreatePosition(p *internal.Position) error {
+	m := t.repo.GetPositions()
+	for _, value := range m {
+		if value.Salary == p.Salary && value.Name == p.Name {
+			return errors.New("position is exists")
+		}
+	}
+	p.ID = uuid.New()
+	t.repo.AddPosition(p)
+	return nil
+}
+
+func (t Serv) CreateEmployee(e *internal.Employee) error {
+	m := t.repo.GetEmployees()
+	for _, value := range m {
+		if value.LasName == e.LasName &&
+			value.FirstName == e.FirstName {
+			return errors.New("employee is exists")
+		}
+	}
+	e.ID = uuid.New()
+	t.repo.AddEmployee(e)
+	return nil
+}
+
+func (t Serv) GetPositions(limit, offset int) ([]internal.Position, error) {
+	m := t.repo.GetPositions()
+	positions := make([]internal.Position, 0)
+	for _, value := range m {
+		positions = append(positions, value)
+	}
+	offset--
+	if float64(len(positions))/float64(limit) < float64(offset) || limit < 1 || offset < 0 {
+		return nil, errors.New("incorrect data")
```

Comment:
could you use more sensible error? `incorrect data` is a very general error which doesn't indicate the root of issue

Reply:
+